src/lib.rs

```diff
@@ -195,20 +216,18 @@ impl SRX {
     ///
     /// Result should be cached instead of calling this repeatedly as it clones the rules.
     pub fn language_rules<S: AsRef<str>>(&self, lang_code: S) -> Rules {
-        let mut rules = Vec::new();
+        let mut rules = Vec::<Rule>::with_capacity(self.map.len());
```

Comment:
The `with_capacity` does not really make sense to me here. The number of rules will almost certainly be much larger than the number of elements in the map (i. e. the number of languages).

Reply:
Lesson learned, no more tired "quick" PRs.